### ISSUE TITLE
`no-undef` fails on a simple array desctructuring

### ISSUE THREAD

**Issue by grabus** [eslint/eslint#2310]
command:
```bash
eslint --no-eslintrc --config ./testconf.json ./test.js
```

test.js:
```javascript
var obj = {
  first: "first",
  second: "second"
};

[obj.first, obj.second] = [obj.second, obj.first];
```

testconf.js:
```javascript
{
  "ecmaFeatures": {
    "destructuring": true
  }
}
```

output:
```
./test.js
  6:5   error  "first" is not defined   no-undef
  6:16  error  "second" is not defined  no-undef

✖ 2 problems (2 errors, 0 warnings)
```

eslint v0.19.0

**Comment by nzakas** [comment]
Thanks, will take a look.

**Comment by jrvidal** [comment]
It may be an EScope bug, I filed estools/escope#62.

**Comment by mysticatea** [comment]
I implemented this: https://github.com/estools/escope/pull/65
I will send a PR to upgrade after released it :)

**Comment by nzakas** [comment]
:+1:

**Comment by samjewell** [comment]
+1 for this
